routes/auth.go

```diff
@@ -40,3 +93,287 @@ func AuthMiddleware() gin.HandlerFunc {
+	AccessToken  tokenResp        `json:"accessToken"`
+}
+
+func getGlobalRole(user *models.User) string {
+	for _, role := range user.GlobalRoles {
+		if role.Role == models.Admin {
+			return "admin"
+		}
+	}
+	return "user"
+}
+
+// Login godoc
+// @Summary    log in using email and password - get refresh tokenResp and initial access tokenResp
+// @Tags       auth
+// @Produce    json
+// @Param      credentials                     body        LoginRequest      true    "User Credentials"
+// @Success    200                             {object}    LoginResponse
+// @Failure    400                             {object}    HTTPErrorResponse
+// @Failure    401                             {object}    HTTPErrorResponse
+// @Failure    500                             {object}    HTTPErrorResponse
+// @Router     /auth/login                     [post]
+func Login(ctx *gin.Context) {
+	log, logCtx := logging.GetRequestLogger(ctx)
+
+	body := LoginRequest{}
+	if err := ctx.ShouldBindJSON(&body); err != nil {
+		log.Warn().Err(err).Msg("validation failed")
+		SendError(ctx, http.StatusBadRequest, err)
+		return
+	}
+
+	user := models.User{
+		UserBase: models.UserBase{
+			Email: body.Email,
+		},
+	}
+
+	db := models.GetDB(logCtx)
+	tx := db.
+		Preload("Organizations").
+		Preload("GlobalRoles").
+		Where(&user).
+		First(&user)
+
+	if err := tx.Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			// we return the same error message in case the email was not found and in case the password was wrong
+			// thus an unauthorized third party is not able to check whether the email is registered with us or not
+			SendError(ctx, http.StatusUnauthorized, errors.New("email or password invalid"))
+		} else {
+			HandleDBError(ctx, logCtx, err)
+		}
+		return
+	}
+
+	if err := bcrypt.CompareHashAndPassword([]byte(user.PwBcrypt), []byte(body.Password)); err != nil {
+		log.
+			Info().
+			Err(err).
+			Str("user_id", user.ID.String()).
+			Msg("Invalid Password")
+		SendError(ctx, http.StatusUnauthorized, errors.New("email or password invalid"))
+		return
+	}
+
+	if len(user.Organizations) == 0 {
+		log.
+			Warn().
+			Str("user_id", user.ID.String()).
+			Msg("no orgs")
+		SendError(ctx, http.StatusBadRequest, errors.New("user is not authorized for any organizations"))
+		return
+	} else if len(user.Organizations) > 1 {
+		log.
+			Warn().
+			Str("user_id", user.ID.String()).
+			Msg("user is in more than one Organization")
+	}
+
+	org := organizationResp{
+		ID:   user.Organizations[0].ID,
+		Role: "user",
+	}
+
+	userResp := userResp{
+		UserBase: user.UserBase,
+		Role:     getGlobalRole(&user),
+	}
+
+	refreshTokenString, refreshTokenClaim, err := issueRefreshToken(&userResp, &org)
+	if err != nil {
+		log.Error().Err(err).Str("refresh_claim", util.Formatted(refreshTokenClaim)).Msg("failed to generate refresh token")
+		SendError(ctx, http.StatusInternalServerError, errors.New("error generating token"))
+		return
+	}
+
+	log.Info().
```

Comment:
We should move the log call for the token issuing to its corresponding function.
In that case, move this log call into the `issueRefreshToken` function.